### ISSUE TITLE
pollTask always runs the first poll?

### ISSUE THREAD

**Issue by dbbk** [ember-lifeline/ember-lifeline#72]
I have a component that updates every 30 seconds. Since adding in `pollTask` my acceptance tests now work and no longer hang because of the nested runloop laters... however they always pause for the first 30 seconds, obviously dramatically slowing down my tests.

Is there any way I can tell Lifeline to simply never run the poll after the component init?

**Comment by scalvert** [comment]
One pattern that works is to configure the interval in which you want to poll. This will reduce the poll time for tests, but can be set to a different value for your prod code.

Does this make sense?

**Comment by dbbk** [comment]
@scalvert yeah that's what I've done at the moment to work around it, but this sounds like a bug with the `pollTask` implementation? My understanding of its use case would be that it never polls/introduces a delay when testing.

**Comment by lolmaus** [comment]
I'm running into this issue too.

I don't quite understand why tests are waiting for `runTask` and `pollTask` to complete. Ain't I supposed to manually control that with `pollTaskFor()`?

Please provide some guidelines on how to use 30 seconds polling without causing every acceptance test to time out.

**Comment by lolmaus** [comment]
PS Making the app tests-aware is a possible workaround, but it's very far from ideal. As is manipulating the app from tests.

Either way you're not actually testing the workflow. You're just hacking the app not to do what it's expected to do. 👎

**Comment by lolmaus** [comment]
Instead, there should be a way to trigger tasks from acceptance tests.

I can imagine something like this:

```
Given there are 20 activity events in the database
And I am on an activity feed page
Then 20 activity events should be visible

Given there is 1 more activity event in the database
When the `activity_feed` timer has triggered
Then 21 activity events should be visible
```

Such test should work without a 30 seconds delay.

Please suggest a way how to make it possible.

**Comment by lolmaus** [comment]
Here's my workaround:

```js
class Foo extends Component {
  isFirstPollComplete = false;

  loadData () {}

  pollData(next: () => any) {
    // This workaround is required to skip the very first poll, so that tests don't lag
    // https://github.com/ember-lifeline/ember-lifeline/issues/72
    if (this.isFirstPollComplete) {
      this.loadData();
      runTask(this, next, Ember.testing ? 0 : 30000);
    } else {
      this.set('isFirstPollComplete', true);
      next();
    }
  }
}
```

`if (this.isFirstPollComplete)` prevents `pollTask` from immediately triggering `runTask`. As a result, `runTask` does *not* happen in tests and tests don't have a delay, letting you test initial state.

Polling can be initiated manually with `pollTaskFor`.

`Ember.testing ? 0 : 30000` prevents polls from wasting 30 seconds of test time. Instead, time can be fast-forwarded with `pollTaskFor`.

**Comment by scalvert** [comment]
Just doing a bit of cleanup on issues in this repo. If this issue is closed in error please reopen!

**Comment by lolmaus** [comment]
The issue is legit, reported by multiple users and never fixed.

**Comment by scalvert** [comment]
Hey @lolmaus, no problem, which is why I added the comment when closing.

Would you mind providing either steps to reproduce or a test that exhibits this issue?

**Comment by lolmaus** [comment]
The two issues mentioned here are rather trivial and I don't think they requires a reproduction for confirmation. They are "by design" and this issue is questioning the design.

1. Acceptance tests are waiting real-life seconds for `runTask` and `pollTask`, which counter-productive, since the whole point of `ember-lifeline` is to be able to advance time manually in tests.

2. `pollTask` immediately executes the task, which makes it impossible to test the initial state.

Both issues can be worked around with boilerplate code. But my and @dbbk's requirements are reasonable and should be common to many developers. Thus, I believe `ember-lifeline` should offer handy out-of-the-box solutions to both problems.

**Comment by lolmaus** [comment]
I believe nobody would want to wait real-life seconds in tests because that would stretch CI from minutes to dozens of minutes.

As for running the task eagerly, `ember-lifeline` could offer a parameter, some configuration option, which would only affect tests, and determine whether the task runs eagerly or expects `pollTaskFor` to be called by hand.

**Comment by scalvert** [comment]
Thanks, @lolmaus. The reason I was asking for steps was so that you could clearly articulate the problem succinctly, so that when I spend my free time investigating at addressing this that time is not wasted. 

I'll spend some time thinking about a solution. Thanks.
